### PR TITLE
Added services for getting users from DB

### PR DESCRIPTION
**Pull request by OrestHaponov**

### CODE REVIEW

File: backend/src/controller/user.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ router.get('/:id', async (req, res, next) => {
 
 // User posts
 
-router.get('/:id/posts', async (req, res, next) => {
+router.get('/:id/getPosts', async (req, res, next) => {
```
Comment (oleksandrstarov) [inline backend/src/controller/user.ts]: no need to use get in route name, keep /posts

File: dto/index.ts [hunk 2]
```diff
@@ -26,27 +25,24 @@ export namespace DTO {
...
   }
 
   export interface IPost {
-    id: number;
+    _id: number;
     body: string;
     date: string;
-    time: string;
-    user: IUserPostDetails;
+    creator: IUserDetailsPost;
```
Comment (oleksandrstarov) [inline dto/index.ts]: usually createdBy field name is used for this

File: dto/index.ts [hunk 3]
```diff
@@ -26,27 +25,24 @@ export namespace DTO {
   }
 
   export interface IFriend {
-    id: number;
-    photoUrl: string;
+    _id: number;
     firstName: string;
     lastName: string;
-    isOnline: boolean;
+    photo: string;
```
Comment (oleksandrstarov) [inline dto/index.ts]: let's use avatar name maybe?

File: frontend/src/components/interfaces/Interface.ts [hunk 4]
```diff
@@ -35,6 +39,7 @@ export interface IUsers {
 export enum dataType {
   filter = 'users/filter',
   users = 'users',
-  posts = 'posts',
+  user = 'user/',
```
Comment (oleksandrstarov) [inline frontend/src/components/interfaces/Interface.ts]: why do we have slash here? I suggest not to use slash here

File: frontend/src/components/pages/profile/Profile.tsx [hunk 5]
```diff
@@ -1,16 +1,41 @@
...
+interface IParamsProps extends RouteComponentProps<MatchParams> { }
+
+const Profile: React.FC<IParamsProps> = (props) => {
   const [userDetails, setUserDetails] = useState(getUserFromStorage());
+  const [userPost, setUserPost] = useState([]);
+  const location = useLocation();
+
+  useEffect(() => {
+    const userId = props.match.params.id;
+    if (userId === undefined) {
```
Comment (oleksandrstarov) [inline frontend/src/components/pages/profile/Profile.tsx]: if (!userId) will be sufficient

File: frontend/src/services/getUserDetails.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+	headers: {
+		'Content-Type': 'application/json'
+	}
+};
+
+const filterUserDetails = async (type: dataType, id: number, data: object = {}) => {
+	let res = null;
+	try {
+		if (Object.keys(data).length === 0 && data.constructor === Object) {
+			res = await axios.get(`${urlApi}${type}${id}`);
```
Comment (oleksandrstarov) [inline frontend/src/services/getUserDetails.ts]: Please check https://www.npmjs.com/package/axios#global-axios-defaults we should set base url globally

File: frontend/src/services/getUserDetails.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import axios from 'axios';
+import { dataType } from '../components/interfaces/Interface';
+
+const urlApi = 'http://127.0.0.1:8000/api/';
+const config = {
+	headers: {
```
Comment (oleksandrstarov) [inline frontend/src/services/getUserDetails.ts]: check indentation

File: frontend/src/services/getUserDetails.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+		'Content-Type': 'application/json'
+	}
+};
+
+const filterUserDetails = async (type: dataType, id: number, data: object = {}) => {
+	let res = null;
+	try {
+		if (Object.keys(data).length === 0 && data.constructor === Object) {
```
Comment (oleksandrstarov) [inline frontend/src/services/getUserDetails.ts]: Object.keys(data).length === 0 >> !Object.keys(data).length

Comment (oleksandrstarov) [inline frontend/src/services/getUserDetails.ts]: data.constructor === Object what we are checking here?

File: frontend/src/services/getUserDetails.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+import { dataType } from '../components/interfaces/Interface';
+
+const urlApi = 'http://127.0.0.1:8000/api/';
+const config = {
+	headers: {
+		'Content-Type': 'application/json'
+	}
+};
+
+const filterUserDetails = async (type: dataType, id: number, data: object = {}) => {
```
Comment (oleksandrstarov) [inline frontend/src/services/getUserDetails.ts]: We do have a get and post here, I cant really get why do we have both options for filter?

File: frontend/src/services/getPostService.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import axios from 'axios';
```
Comment (oleksandrstarov) [inline frontend/src/services/getPostService.ts]: this file is same as one below... shouldnt be like that, smells like OOD principles are not followed here

File: frontend/src/components/shared/userPhoto/UserPhoto.tsx [hunk 11]
```diff
@@ -1,12 +1,11 @@
 import React from 'react';
 import { IUserPhotoProps } from '../../interfaces/Interface';
-import Avatar from '../../mocks/user.png';
 import './userPhoto.scss';
 
 export const UserPhoto: React.FC<IUserPhotoProps> = (props) => {
```
Comment (oleksandrstarov) [inline frontend/src/components/shared/userPhoto/UserPhoto.tsx]: as a rule - it is easier to see what we are using from props
React.FC<IUserPhotoProps> = ({ photo, isOnline }) => {

File: frontend/src/components/shared/userPhoto/UserPhoto.tsx [hunk 12]
```diff
@@ -1,12 +1,11 @@
...
   return (
     <div className="user-photo">
-      <img src={Avatar} alt="User avatar" />
+      <img src={props.photo} alt="User avatar" />
       {props.isOnline ? <span className="online-status"></span> : null}
```
Comment (oleksandrstarov) [inline frontend/src/components/shared/userPhoto/UserPhoto.tsx]: {isOnline && <span className="online-status"></span>}

File: frontend/src/components/shared/userCard/UserCard.tsx [hunk 13]
```diff
@@ -64,18 +63,14 @@ export const UserCard: React.FC<MainInfoProps> = ({ mainInfo }, shadowIntensity)
...
+            {loginedUser.id
+              ? null
               : <div className="buttons-action">
                 {mainInfo.isFriend
                   ? null
-                  : <Link to="/" onClick={sendFriendRequest} className="add-friend">
-                    add friend
-              </Link>
+                  : <Link to="/" onClick={sendFriendRequest} className="add-friend">add friend</Link>
                 }
```
Comment (oleksandrstarov) [inline frontend/src/components/shared/userCard/UserCard.tsx]: `{!mainInfo.isFriend && <Link to="/" onClick={sendFriendRequest} className="add-friend">add friend</Link>}`

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

**Comment by OrestHaponov** [comment]
@oleksandrstarov Updated services for frontend please check

File: frontend/src/components/pages/profile/Profile.tsx [hunk 14]
```diff
@@ -9,8 +10,29 @@ import FriendsList from './components/friendsList/FriendsList';
...
+
+interface MatchParams {
+  id: number & string;
+}
+
+interface IParamsProps extends RouteComponentProps<MatchParams> { }
+
+const Profile: React.FC<IParamsProps> = (props) => {
   const [userDetails, setUserDetails] = useState(getUserFromStorage());
+  const [userPost, setUserPost] = useState([]);
```
Comment (oleksandrstarov) [inline frontend/src/components/pages/profile/Profile.tsx]: minor one - use plurals for array - userPosts in this case setUserPosts

File: frontend/src/services/apiService.ts [hunk 15]
```diff
@@ -1,20 +1,14 @@
...
-  }
-};
+axios.defaults.baseURL = 'http://127.0.0.1:8000/api/';
 
 const apiService = async (type: dataType, data: object = {}) => {
   let res = null;
   try {
     if (Object.keys(data).length === 0 && data.constructor === Object) {
-      res = await axios.get(`${urlApi}${type}`);
+      res = await axios.get(`${axios.defaults.baseURL}${type}`);
```
Comment (oleksandrstarov) [inline frontend/src/services/apiService.ts]: with setting defaults you can use axios.get('/user/12345')

File: frontend/src/components/shared/userCard/UserCard.tsx [hunk 16]
```diff
@@ -64,18 +63,11 @@ export const UserCard: React.FC<MainInfoProps> = ({ mainInfo }, shadowIntensity)
                 );
               })}
             </div>
-            {loginedUser.id === mainInfo.id ?
-              null
+            {loginedUser.id
+              ? null
```
Comment (oleksandrstarov) [inline frontend/src/components/shared/userCard/UserCard.tsx]: same comments here >> 

`{ !loginedUser.id && <....>}`

**Review by oleksandrstarov — APPROVED** [review]
Please check comments, looks good otherwise

**Pull request merged**